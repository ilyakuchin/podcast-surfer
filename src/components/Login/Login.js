import React, { useState } from "react";

export default function Login() {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  function sbmt(e) {
    e.preventDefault();
    console.log("login");
  }
  function handleUsernameChange(e) {
    setUsername(e.target.value);
  }

  function handlePasswordChange(e) {
    setPassword(e.target.value);
  }

  return (
    <div>
      <form>
        <label>Username</label>
        <input
          type="text"
          value={username}
          onChange={handleUsernameChange}
          placeholder="username"
          name="uname"
          required
        />

        <label>Password</label>
        <input
          type="password"
          value={password}
          onChange={handlePasswordChange}
          placeholder="password"
          name="pswrd"
          required
        />
        <input type="submit" value="Submit" onClick={sbmt} />
      </form>
    </div>
  );
}
