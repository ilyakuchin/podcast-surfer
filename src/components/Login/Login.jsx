import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import {
  setUsername,
  setPassword,
  login
} from '../../redux/actions/UserInfo/userInfo';

export function Login({
  username,
  password,
  validationErrorMessage,
  setUsernameConnect,
  setPasswordConnect,
  loginConnect
}) {
  return (
    <div>
      <h2>LOGIN</h2>
      <Link to='/signup'>Sign Up</Link>
      <div>{validationErrorMessage}</div>
      <form>
        <div>Username</div>
        <input
          type='text'
          value={username}
          onChange={e => setUsernameConnect(e.target.value)}
          placeholder='username'
          name='uname'
          required
        />

        <div>Password</div>
        <input
          type='password'
          value={password}
          onChange={e => setPasswordConnect(e.target.value)}
          placeholder='password'
          name='pswrd'
          required
        />
        <input
          type='submit'
          value='Submit'
          onClick={e => loginConnect(e, username, password)}
        />
      </form>
    </div>
  );
}

Login.propTypes = {
  username: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  validationErrorMessage: PropTypes.string.isRequired,
  setUsernameConnect: PropTypes.func.isRequired,
  setPasswordConnect: PropTypes.func.isRequired,
  loginConnect: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    username: state.userInfo.username,
    password: state.userInfo.password,
    validationErrorMessage: state.userInfo.validationErrorMessage
  };
};

const dispatchToProps = dispatch => {
  return {
    setUsernameConnect: username => dispatch(setUsername(username)),
    setPasswordConnect: password => dispatch(setPassword(password)),
    loginConnect: (e, username, password) => {
      e.preventDefault();
      dispatch(login(username, password));
    }
  };
};

const ConnectedLogin = connect(mapStateToProps, dispatchToProps)(Login);

export default ConnectedLogin;
