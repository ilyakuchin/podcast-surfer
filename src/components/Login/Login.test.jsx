import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Login } from './Login';

configure({ adapter: new Adapter() });

describe('Test Login component', () => {
  test('should render correctly', () => {
    expect(
      toJson(
        shallow(
          <Login
            username='username'
            password='password'
            validationErrorMessage='errorMessage'
          />
        )
      )
    ).toMatchSnapshot();
  });
});
