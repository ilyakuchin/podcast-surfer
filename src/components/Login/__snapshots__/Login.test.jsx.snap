// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login component should render correctly 1`] = `
<div>
  <h2>
    LOGIN
  </h2>
  <Link
    to="/signup"
  >
    Sign Up
  </Link>
  <form>
    <div>
      Username
    </div>
    <input
      name="uname"
      onChange={[Function]}
      placeholder="username"
      required={true}
      type="text"
      value=""
    />
    <div>
      Password
    </div>
    <input
      name="pswrd"
      onChange={[Function]}
      placeholder="password"
      required={true}
      type="password"
      value=""
    />
    <input
      onClick={[Function]}
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
